projects/feature-activation/0003-explorer-uis.md

```diff
@@ -0,0 +1,72 @@
+- Feature Name: Feature Activation Explorer UIs
+- Start Date: 2023-07-03
+- Author: Gabriel Levcovitz <<gabriel@hathor.network>>
+
+# Summary
+[summary]: #summary
+
+Given the [Feature Activation for Blocks RFC](./0001-feature-activation-for-blocks.md), a mechanism is provided to deploy new features in the Hathor network conditionally on some criteria. It defined two new Explorer user interfaces: a new page containing a table of all features in the Feature Activation process and their metadata, and a new panel in the Block interface displaying the signal bits and respective features for that block. This RFC details those interfaces.
+
+# Motivation
+[motivation]: #motivation
+
+There should be an easy way for users to follow the Feature Activation process and each feature's current state. Using the Explorer is ideal for this.
+
+# Guide-level explanation
+[Guide-level explanation]: #guide-level-explanation
+
+This section defines what the interfaces will contain and provides a simple mockup for them.
+
+## New Feature Activation interface
+
+There must be a new option in the Tools dropdown, at the Explorer header. When clicked, it will open a new page specific for Feature Activation information. It will include a table as described in [this section](https://github.com/HathorNetwork/rfcs/blob/master/projects/feature-activation/0001-feature-activation-for-blocks.md#explorer-user-interface) of the original RFC, containing all Criteria for each feature, and also it's current state and acceptance percentage.
+
+Also, there must be an indication on the block used to retrieve the "current" information, that is, the best block when the request was made. Its height should be displayed and there should be a link to its page (this is not shown in the mockup below).
+
+### Mockup
+
+![feature_activation.png](0003-images%2Ffeature_activation.png)
+
+## New panel in Block interface
+
+There must be a new panel in the Block interface containing the Feature Activation information for that block, that is, a table with the block's signal value for each feature, the corresponding bit and feature name, and the feature state at that point.
```

Comment:
Signal `0` or `1` is if the feature is active or not in this block? If that's the case, I think I prefer an icon with a check in the one active (and maybe an X in the others, although I usually prefer marking only the ones active to make the UI cleaner).

Do we have a fixed list of possible states? If that's the case, I think we should have a pretty state to show in this page, so we don't show the string with underline.

What do you think?

Reply:
Added the signal suggestion in https://github.com/HathorNetwork/rfcs/pull/60/commits/47b161c998b637f8427fe093426b93022078fd3f.

> Do we have a fixed list of possible states? If that's the case, I think we should have a pretty state to show in this page, so we don't show the string with underline.

We do have a list of fixed states, but what do you mean by pretty state?

Reply:
I think he means this:

![image](https://github.com/HathorNetwork/rfcs/assets/3586068/8f998e13-16bd-4f32-b963-1bdfce88fedf)

And I agree, can we map those to "pretty" values? Like "Started", "Must Signal", ...

Reply:
Ahh, makes sense. Ok, I'll make sure to do that